pkg/combustion/kubernetes_test.go

```diff
@@ -821,3 +1511,191 @@ func TestConfigureKubernetes_SuccessfulRKE2ServerWithManifests(t *testing.T) {
 	assert.Contains(t, contents, "name: my-nginx")
 	assert.Contains(t, contents, "image: nginx:1.14.2")
 }
+
+func TestKubernetesVIPManifestValidIPV4(t *testing.T) {
+	k8s := &image.Kubernetes{
+		Version: "v1.30.3+rke2r1",
+		Network: image.Network{
+			APIVIP4: "192.168.1.1",
+		},
+	}
+
+	manifest, err := kubernetesVIPManifest(k8s)
+	require.NoError(t, err)
+
+	assert.Contains(t, manifest, "- 192.168.1.1/32")
+	assert.Contains(t, manifest, "- name: rke2-api")
+	assert.NotContains(t, manifest, "ipFamilies:\n      - IPv6")
+	assert.NotContains(t, manifest, "ipFamilyPolicy: SingleStack")
```

Comment:
We can have assertions for both `SingleStack` and `RequireDualStack` in all cases in order to easily make the difference. All templated values under the different conditions should be tested.